js/piwik.js

```diff
@@ -2605,10 +2609,34 @@ if (typeof Piwik !== 'object') {
                 callback();
             }
 
+            /**
+             * Send each bulk tracks to the server
+             * @return {void}
+             */
+            function sendBulkTrackingRequests() {
+
+                // Work on a local copy
+                var tmp = JSON.parse(JSON.stringify(bulkTrackingCollection));
```

Comment:
as this method will be called every 300ms by default it would be much faster to first check for `if (!sendBulkTrackingRequests || !sendBulkTrackingRequests.length) { return; }` instead of serializing it every 300ms and then check if there is actually an item.